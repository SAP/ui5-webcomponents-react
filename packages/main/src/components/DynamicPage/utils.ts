export const DynamicPageCssVariables = {
  headerDisplay: '--_ui5wcr_DynamicPage_header_display',
  titleFontSize: '--_ui5wcr_DynamicPage_title_fontsize'
};
