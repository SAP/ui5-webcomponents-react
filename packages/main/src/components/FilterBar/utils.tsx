import { cloneElement } from 'react';

const inputTagNames = new Set([
  'UI5-COMBOBOX',
  'UI5-DATE-PICKER',
  'UI5-DATERANGE-PICKER',
  'UI5-DATETIME-PICKER',
  'UI5-DURATION-PICKER',
  'UI5-INPUT',
  'UI5-MULTI-INPUT',
  'UI5-RATING-INDICATOR',
  'UI5-STEP-INPUT',
  'UI5-TIME-PICKER'
]);

export const filterValue = (ref, child) => {
  const tagName = ref.tagName;
  let filterItemProps = {};
  if (inputTagNames.has(tagName)) {
    filterItemProps = { value: ref.value };
  }
  if (tagName === 'UI5-SELECT' || tagName === 'UI5-MULTI-COMBOBOX') {
    const selectedIndices = Array.from(ref.children as HTMLCollectionOf<any>)
      .map((item, index) => (item.selected ? index : false))
      .filter((el) => el !== false);
    const selectedIndicesSet = new Set(selectedIndices);
    const options = child.props.children.props.children?.map((item, index) => {
      if (selectedIndicesSet.has(index)) {
        return cloneElement(item, { selected: true });
      }
      return cloneElement(item, { selected: false });
    });
    filterItemProps = { children: options };
  }
  if (tagName === 'UI5-SWITCH' || tagName === 'UI5-CHECKBOX') {
    filterItemProps = { checked: ref.checked };
  }
  return filterItemProps;
};

export const renderSearchWithValue = (renderSearchElement, searchValue) => {
  return cloneElement(renderSearchElement, { value: searchValue ?? renderSearchElement?.props?.value });
};
