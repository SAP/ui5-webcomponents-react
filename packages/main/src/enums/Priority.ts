// Generated file - do not change manually!

import Priority from '@ui5/webcomponents/dist/types/Priority.js';

export { Priority };
