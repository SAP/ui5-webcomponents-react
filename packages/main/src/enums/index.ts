// This is an autogenerated file, please do not modify this file manually.
// In case you added a new file to the /enums folder, pleas rerun the scripts/create-enum-export.js script.

export * from './AnalyticalTableScrollMode';
export * from './AvatarColorScheme';
export * from './AvatarGroupType';
export * from './AvatarShape';
export * from './AvatarSize';
export * from './BarDesign';
export * from './BreadcrumbsDesign';
export * from './BreadcrumbsSeparatorStyle';
export * from './BusyIndicatorSize';
export * from './ButtonDesign';
export * from './CalendarSelection';
export * from './CalendarSelectionMode';
export * from './CalendarType';
export * from './CarouselArrowsPlacement';
export * from './ContentDensity';
export * from './DeviationIndicator';
export * from './FCLLayout';
export * from './FlexBoxAlignItems';
export * from './FlexBoxDirection';
export * from './FlexBoxJustifyContent';
export * from './FlexBoxWrap';
export * from './GlobalStyleClasses';
export * from './GridPosition';
export * from './IllustrationMessageType';
export * from './IndicationColor';
export * from './InputType';
export * from './LinkDesign';
export * from './ListGrowingMode';
export * from './ListItemType';
export * from './ListMode';
export * from './ListSeparators';
export * from './LoaderType';
export * from './MediaGalleryItemLayout';
export * from './MediaGalleryLayout';
export * from './MediaGalleryMenuHorizontalAlign';
export * from './MediaGalleryMenuVerticalAlign';
export * from './MessageBoxActions';
export * from './MessageBoxTypes';
export * from './MessageStripDesign';
export * from './ObjectPageMode';
export * from './PageBackgroundDesign';
export * from './PanelAccessibleRole';
export * from './PopoverHorizontalAlign';
export * from './PopoverPlacementType';
export * from './PopoverVerticalAlign';
export * from './Priority';
export * from './SemanticColor';
export * from './SideContentFallDown';
export * from './SideContentPosition';
export * from './SideContentVisibility';
export * from './Size';
export * from './SwitchDesign';
export * from './TabLayout';
export * from './TableGrowingMode';
export * from './TableMode';
export * from './TableRowType';
export * from './TableScaleWidthMode';
export * from './TableSelectionBehavior';
export * from './TableSelectionMode';
export * from './TableVisibleRowCountMode';
export * from './TabsOverflowMode';
export * from './TextAlign';
export * from './Themes';
export * from './TimelineLayout';
export * from './TitleLevel';
export * from './ToastPlacement';
export * from './ToolbarDesign';
export * from './ToolbarStyle';
export * from './UploadState';
export * from './ValueColor';
export * from './ValueState';
export * from './VerticalAlign';
export * from './WrappingType';
