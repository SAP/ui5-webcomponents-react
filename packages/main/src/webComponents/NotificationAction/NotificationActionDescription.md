The `NotificationAction` represents an abstract action, used in the `NotificationListItem` and the `NotificationListItemBase` items
