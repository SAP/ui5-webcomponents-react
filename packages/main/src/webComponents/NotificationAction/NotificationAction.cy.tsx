import { cypressPassThroughTestsFactory } from '@/cypress/support/utils';
import { NotificationAction } from './index';

describe('NotificationAction', () => {
  it('Basic Test (generated)', () => {
    cy.mount(<NotificationAction />);
    cy.get('[ui5-notification-action]').should('exist');
  });

  cypressPassThroughTestsFactory(NotificationAction);
});
