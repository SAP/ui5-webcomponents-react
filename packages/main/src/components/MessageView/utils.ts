import iconAlert from '@ui5/webcomponents-icons/dist/alert.js';
import iconError from '@ui5/webcomponents-icons/dist/error.js';
import iconInformation from '@ui5/webcomponents-icons/dist/information.js';
import iconSysEnter from '@ui5/webcomponents-icons/dist/sys-enter-2.js';
import { ValueState } from '../../enums/ValueState';

export const getIconNameForType = (type: ValueState | keyof typeof ValueState): string => {
  switch (type) {
    case ValueState.Error:
    case 'Error':
      return iconError;
    case ValueState.Success:
    case 'Success':
      return iconSysEnter;
    case ValueState.Warning:
    case 'Warning':
      return iconAlert;
    default:
      return iconInformation;
  }
};
