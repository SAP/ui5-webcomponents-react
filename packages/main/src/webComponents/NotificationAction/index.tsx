import { ButtonDesign } from '../../enums';
import { CommonProps } from '../../interfaces/CommonProps';
import { Ui5DomRef } from '../../interfaces/Ui5DomRef';
import { withWebComponent } from '../../internal/withWebComponent';

import '@ui5/webcomponents-fiori/dist/NotificationAction.js';

interface NotificationActionAttributes {
  /**
   * Defines the action design.
   *
   * **Note:**
   *
   * *   `Default`
   * *   `Emphasized`
   * *   `Positive`
   * *   `Negative`
   * *   `Transparent`
   */
  design?: ButtonDesign | keyof typeof ButtonDesign;
  /**
   * Defines if the action is disabled.
   *
   * **Note:** a disabled action can't be pressed or focused, and it is not in the tab chain.
   */
  disabled?: boolean;
  /**
   * Defines the `icon` source URI.
   *
   * **Note:** SAP-icons font provides numerous built-in icons. To find all the available icons, see the <ui5-link target="_blank" href="https://openui5.hana.ondemand.com/test-resources/sap/m/demokit/iconExplorer/webapp/index.html" class="api-table-content-cell-link">Icon Explorer</ui5-link>.
   */
  icon?: string;
  /**
   * Defines the text of the `NotificationAction`.
   */
  text?: string;
}

export interface NotificationActionDomRef extends NotificationActionAttributes, Ui5DomRef {}

export interface NotificationActionPropTypes extends NotificationActionAttributes, CommonProps {}

/**
 * The `NotificationAction` represents an abstract action, used in the `NotificationListItem` and the `NotificationListItemBase` items
 *
 * <ui5-link href="https://sap.github.io/ui5-webcomponents/playground/components/NotificationAction" target="_blank">UI5 Web Components Playground</ui5-link>
 */
const NotificationAction = withWebComponent<NotificationActionPropTypes, NotificationActionDomRef>(
  'ui5-notification-action',
  ['design', 'icon', 'text'],
  ['disabled'],
  [],
  []
);

NotificationAction.displayName = 'NotificationAction';

NotificationAction.defaultProps = {
  design: ButtonDesign.Transparent
};

export { NotificationAction };
