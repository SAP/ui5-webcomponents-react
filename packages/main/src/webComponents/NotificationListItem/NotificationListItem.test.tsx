import { render } from '@shared/tests';
import { createCustomPropsTest, createWaitForDefineTest } from '@shared/tests/utils';
import { NotificationListItem } from './index';
import { NotificationAction } from '../NotificationAction';
import { Priority } from '../../enums/Priority';
import React from 'react';

describe('NotificationListItem', () => {
  test('Basic Test (generated)', () => {
    const { asFragment } = render(
      <NotificationListItem
        titleText="My Notification"
        showClose
        priority={Priority.High}
        actions={<NotificationAction text="My Action" />}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  createCustomPropsTest(NotificationListItem);
  createWaitForDefineTest(NotificationListItem);
});
