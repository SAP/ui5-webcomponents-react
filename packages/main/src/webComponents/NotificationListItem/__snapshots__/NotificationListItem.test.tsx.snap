// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationListItem Basic Test (generated) 1`] = `
<DocumentFragment>
  <ui5-li-notification
    busy-delay="1000"
    priority="High"
    show-close="true"
    title-text="My Notification"
    ui5-li-notification=""
  >
    <ui5-notification-action
      design="Transparent"
      slot="actions"
      text="My Action"
      ui5-notification-action=""
    />
  </ui5-li-notification>
</DocumentFragment>
`;
